src/libostree/ostree-repo.c

```diff
@@ -2823,12 +2823,10 @@ reload_core_config (OstreeRepo          *self,
   }
 
   {
-    if (g_key_file_has_key (self->config, "core", "min-free-space-size", NULL) &&
-        g_key_file_has_key (self->config, "core", "min-free-space-percent", NULL))
-      {
-        return glnx_throw (error, "min-free-space-percent and min-free-space-size are mutually exclusive");
```

Comment:
>  Why not just change this to a g_warning and change the subsequent `else if` to `if`?

... and subsequent `else` to `if` too? (We need to parse both config options if present)

> The code was already handling the fallbacks correctly

Yes, but it was also parsing any one of these options, not both. That's why we need to split out fallback (checking none of the key is there)

> ... all that's desired is to not error if both options are set

We can't just lookup respective keys and decide. We need to parse them first.
Consider both keys are present, we need to parse them first because even any one of them is zero, we consider  that key is disabled by user (hence it's a valid case here and shouldn't warn). This was suggested by pwithnall and I agree with him [1]

>  It also wouldn't warn if someone set min-free-space-percent=0 and min-free-space-size=0

Exactly. I believe this is the correct behavior, as even if any one of them is `0`, we consider it a valid case (and doesn't warn).

[1] https://phabricator.endlessm.com/T13698#610153

Reply:
OK, now I understand why you're going to all the trouble. Personally I think the warning should always happen if both `min-free-space-size` and `min-free-space-percent`, but no big deal.

On the other hand, you've changed 2 behaviors in this commit:

1. ostree will now warn instead of error if both options are set
2. If `min-free-space-size=0`, then ostree will use `min-free-space-percent. Previously if `min-free-space-size` was defined, then `min-free-space-percent` would be ignored.

It would be nice if you could break this into 2 commits or explain that more clearly in the commit message.